src/org/labkey/test/tests/AdminConsoleNavigationTest.java

```diff
@@ -71,7 +108,134 @@ public void testAdminNavTrails()
+    {
+        Set<String> ignoredLinksNonAdmin = Collections.newSetFromMap(new CaseInsensitiveHashMap<>());
+        ignoredLinksNonAdmin.addAll(List.of(
+                "Merge sync admin",     //Can be accessed by non admin
+                "Dump Heap",            // Undesired consequences
+                "Reset Site Errors",    // Undesired consequences
+                "Running Threads",      // Undesired consequences
+                "Credits"               //Can be accessed by non admin
+        ));
+
+        Set<String> ignoredLinksAdmin = Collections.newSetFromMap(new CaseInsensitiveHashMap<>());
+        ignoredLinksAdmin.addAll(List.of(
+                "Dump Heap",            // Undesired consequences
+                "Reset Site Errors",    // Undesired consequences
+                "Running Threads"      // Undesired consequences
+        ));
+        ShowAdminPage adminConsole = goToAdminConsole();
+        List<WebElement> adminLinks = adminConsole.getAllAdminConsoleLinks();
+        Map<String, String> linkHrefs = new LinkedHashMap<>();
+        for (WebElement link : adminLinks)
+            linkHrefs.put(link.getText(), link.getAttribute("href"));
+
+        log("Verifying links can be accessed by admin");
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (ignoredLinksAdmin.contains(link.getKey()))
+                TestLogger.log("Skipping admin link: " + link.getKey());
+            else
+            {
+                log("Verifying link " + link.getKey() + " with URL " + link.getValue());
+                verifyLinks(link.getKey(), link.getValue(), 200);
+            }
+        }
+
+        log("Verifying links cannot be accessed by non admin");
+        goToHome();
+        impersonate(NON_ADMIN);
+        for (Map.Entry<String, String> link : linkHrefs.entrySet())
+        {
+            if (ignoredLinksNonAdmin.contains(link.getKey()))
+                TestLogger.log("Skipping admin link: " + link.getKey());
+            else
+            {
+                log("Verifying link " + link.getKey() + " with URL " + link.getValue());
+                verifyLinks(link.getKey(), link.getValue(), 403);
+            }
+        }
+        goToHome();
+        stopImpersonating();
+    }
+
+    private void verifyReadOnlyLinks(String text, String url)
+    {
+        beginAt(url);
+        Assert.assertEquals("URL " + url + " is broken for " + text, 200, getResponseCode());
+        Locator.XPathLocator updateButtonLoc = Locator.XPathLocator.union(
+                Locator.linkWithText("Submit"), Locator.linkWithText("Save"), Locator.linkWithText("Update")).notHidden();
+        assertFalse("Either Save/Submit/Update button exists in the page " + text, isElementPresent(updateButtonLoc));
+    }
+
+    private void verifyPostCommandFails(String url)
+    {
+        String[] actionController = url.substring(url.lastIndexOf('/') + 1).split("-");
```

Comment:
This isn't a valid way to extract the action and controller from a URL; some controllers include a hyphen (e.g. `mini-profiler-manage.view`.
`new Crawler.ControllerActionId(url)` will correctly parse all of the url variations we support. It will also give you the container path, which could be used below (though I would expect it to always be root).